WorkflowTesting/Sources/RenderTesterResult.swift

```diff
@@ -41,6 +43,22 @@ public struct RenderTesterResult<WorkflowType: Workflow> {
         return self
     }
 
+    /// Exhaustive state testing against the initial state.
+    /// - Parameters:
+    ///   - changes: A function that receives the initial state
+    ///   and is expected to mutate it to match the new state.
+    @discardableResult
+    public func assertState(
```

Comment:
i wonder if naming like `assertStateModifications`, `assertModifiedState`, `assertTransformedState`, or something along those lines might be a bit more clear. since there's already an `assert(state:)` method, might be good to make the difference a bit more obvious.

Reply:
`assertStateModifications` is pretty close to what I have now if you read the argument as part of the name

```
assertState(changes:)
```

Reply:
come to think of it `assert(stateChanges:)` or `assert(stateModifications:)` might be a closer match to the existing pattern. wdyt?

Reply:
i guess my personal preference is just to try and avoid having a number of different methods that end up reading like `assert { ... }` with different closure types, as i find that harder to search for, and harder to understand when visually scanning.